.github/workflows/pr-environment-update.yml

```diff
@@ -0,0 +1,50 @@
+name: PR Environment Update
+run-name: Update PR Environment ${{ github.event.number }}
+on:
+  workflow_call:
+    inputs:
+      app_name:
+        required: true
+        type: string
+      environment:
+        required: true
+        type: string
+jobs:
+  build-and-publish:
+    name: " " # GitHub UI is noisy when calling reusable workflows, so use whitespace for name to reduce noise
+    uses: ./.github/workflows/build-and-publish.yml
+    with:
+      app_name: ${{ inputs.app_name }}
+      ref: ${{ github.sha }}
```

Comment:
I'm not positive it matters for this case, but wanted to flag that you're using `github.sha` here, but `database-migrations.yml` workflow (I think the only other time we call `build-and-publish.yml`) uses `github.ref`: https://github.com/navapbc/template-infra/blob/4c396474e12f36c3a5166370a10cfafca5df0372/.github/workflows/database-migrations.yml#L23

I looked at https://docs.github.com/en/actions/learn-github-actions/contexts#github-context but still found it hard to determine whether it matters here.

Reply:
It can be either github.ref or github.sha here, since [build-and-publish gets the sha hash as the first job](https://github.com/navapbc/template-infra/blob/main/.github/workflows/build-and-publish.yml#L27-L41), but below when we do the apply we need to pass the sha for the image_tag (passed into update-pr-environment on line 48), since the ref won't work there. Since it needs to be the sha there, I figured it'd be clearer to use the sha in both places so it's clear that it's the same thing.